src/legacy/core_plugins/kibana/public/discover/np_ready/angular/discover.js

```diff
@@ -188,23 +178,149 @@ function discoverController(
   $scope,
   $timeout,
   $window,
-  AppState,
   Promise,
   config,
   kbnUrl,
   localStorage,
-  uiCapabilities,
-  getAppState,
-  globalState
+  uiCapabilities
 ) {
-  const filterStateManager = new FilterStateManager(globalState, getAppState, filterManager);
+  const subscriptions = new Subscription();
+  const $fetchObservable = new Subject();
+  let inspectorRequest;
+  const savedSearch = $route.current.locals.savedObjects.savedSearch;
+  if ($route.current.locals.savedObjects.savedQuery) {
+    $scope.savedQuery = $route.current.locals.savedObjects.savedQuery;
+  }
+
+  $scope.searchSource = savedSearch.searchSource;
+  $scope.indexPattern = resolveIndexPatternLoading();
+  $scope.timeRangeObj = {
+    ..._.cloneDeep(timefilter.getTime()),
+  };
```

Comment:
Could it be just:
`$scope.timeRangeObj = timefilter.getTime()`
Didn't find a necessity to use deep clone and desctructuring (even if deep clone is necessary, dectructuring is redundant)

Reply:
👍 done